src/csharp/Microsoft.Spark.Worker/Command/SqlCommandExecutor.cs

```diff
@@ -165,8 +165,33 @@ private void WriteOutput(Stream stream, IEnumerable<object> rows, int sizeHint)
             if (s_outputBuffer == null)
                 s_outputBuffer = new byte[sizeHint];
 
+            // Cast GenericRow to object if needed.
+            var newRows = new List<object>();
+            if (rows.FirstOrDefault() is GenericRow)
+            {
+                for (int i = 0; i < rows.Count(); ++i)
+                {
+                    object[] cols = (rows.ElementAt(i) as GenericRow).Values;
+                    if (cols.FirstOrDefault() is GenericRow)
+                    {
+                        object[] newCols = new object[cols.Length];
```

Comment:
Don't you need to do this recursively? What happens you have a column with struct type in multiple levels?

Reply:
Instead of making modifications here, how about registering a custom pickler for GenericRow.

Reply:
@suhsteve I am a bit confused. Since this is used for `dumps`, wouldn't this require changes on Spark side? Can you provide snippet of what you are proposing?

Reply:
I haven't tested it yet, but I'm thinking something like:

```csharp
public class GenericRowPickler : IObjectPickler {
    public void pickle(object o, Stream outs, Pickler currentPickler) {
        GenericRow row = (GenericRow)o;
        currentPickler.save(row.Values);
    }
}
```

And then register the customer pickler somewhere like:
```csharp
Pickler.registerCustomPickler(typeof(GenericRow), new GenericRowPickler());
```

Reply:
Cool! @elvaliuliuliu, can you try this? This should work.

Reply:
Thanks for the suggestion! I have made the changes accordingly. Let me know if any feedback. Thanks a lot!